### PR TITLE
Issue #223692 feat : Course planner UI updated for state, bords, subject screens

### PR DESCRIPTION
**Pull request by rahulg1254**

### CODE REVIEW

File: src/components/CardItem.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      onClick={onClick}
+    >
+      <Box sx={{ display: "flex", alignItems: "center", gap: "8px" }}>
+        <FolderOutlinedIcon />
+        <Typography variant="h6">{card.state}</Typography>
+      </Box>
+      <Box sx={{ display: "flex", alignItems: "center", gap: "16px" }}>
+        <CustomStepper completedSteps={card.boardsUploaded} />
+        <Typography sx={{ fontSize: "14px" }}>
+          {card.boardsUploaded} / {card.totalBoards} {"boards fully uploaded"}
```
Comment (itsvick) [inline src/components/CardItem.tsx:52]: @rahulg1254 translate this

File: src/components/FileUploadDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  onClose,
+  onFileChange,
+  selectedFile,
+  onRemoveFile,
+  onUpload,
+}) => {
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="sm" fullWidth>
+      <DialogTitle sx={{ position: "relative" }}>
+        <Typography variant="h2">Import Planner</Typography>
```
Comment (itsvick) [inline src/components/FileUploadDialog.tsx:35]: @rahulg1254 translate

File: src/components/HeaderComponent.tsx [hunk 3]
```diff
@@ -202,30 +202,28 @@ const HeaderComponent = ({
...
+              onChange={handleFilterChange}
+              displayEmpty
+              style={{
+                borderRadius: "8px",
+                height: "40px",
+                fontSize: "14px",
+              }}
+            >
+              <MenuItem value="All">
+                <em>All</em>
```
Comment (itsvick) [inline src/components/HeaderComponent.tsx:217]: @rahulg1254 translate

File: src/components/HeaderComponent.tsx [hunk 4]
```diff
@@ -202,30 +202,28 @@ const HeaderComponent = ({
...
+              </MenuItem>
+              {Filter?.map((filter, index) => (
+                <MenuItem value={filter} key={index}>
```
Comment (itsvick) [inline src/components/HeaderComponent.tsx:220]: @rahulg1254 avoid using index as key

File: src/components/CardItem.tsx (hunk 1 above)
Comment (rahulg1254) [reply src/components/CardItem.tsx:52]: Resolved

File: src/components/HeaderComponent.tsx (hunk 3 above)
Comment (rahulg1254) [reply src/components/HeaderComponent.tsx:217]: Resolved

File: src/components/FileUploadDialog.tsx (hunk 2 above)
Comment (rahulg1254) [reply src/components/FileUploadDialog.tsx:35]: Resolved

**Pull request merged**